docs/docs/development/hardware-integration/index.mdx

```diff
@@ -0,0 +1,122 @@
+---
+title: Hardware Integration
+sidebar_label: Overview
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+This section of the documentation describes steps necessary to get ZMK running on a keyboard.
+Please see pages in the sidebar for guides and reference that describe different aspects of this integration.
+
+The foundational elements needed to get a specific keyboard working with ZMK can be broken down into:
+
+- A [physical layout](physical-layouts.md) that describes the electrical and physical structure of the keyboard, referring to:
+  - A [kscan driver](../../config/kscan.md), which most frequently uses `compatible = "zmk,kscan-gpio-matrix"` for GPIO matrix based keyboards, or `compatible = "zmk,kscan-gpio-direct"` for direct wires.
+  - A [matrix transform](../../config/layout.md), which defines how the kscan row/column events are translated into logical "key positions".
+  - An [optional description](physical-layouts.md#optional-keys-property) of physical key positions and sizes, in order to visualize the keyboard accurately in [ZMK Studio](../../features/studio.md).
+- A [keymap](../../keymaps/index.mdx), which binds each key position to a behavior, e.g. key press, mod-tap, momentary layer, in a set of layers.
+- Other, optional configuration items to support features such as encoders or lighting systems.
+
+These core architectural elements are defined per-keyboard, and _where_ they are defined depends on the specifics of how that keyboard is structured.
+
+## Boards & Shields
+
+ZMK uses the Zephyr concepts of "boards" and "shields" to refer to different parts of a keyboard build, that in turn get combined during a firmware build.
+This provides the modularity to be able to use [composite keyboards](#composite-keyboards) with different compatible controllers.
+Also see the "Why boards and shields" question in the [FAQ](../../faq.md#why-boards-and-shields-why-not-just-keyboard) for a longer explanation.
```

Comment:
I'm not a fan of the circular linking happening here. I think we should just pull all of that out of the FAQ and move it here, rather than splitting it in half.

Reply:
Moved it all here, made it less verbose, left a header with mostly just a link in FAQ.